### ISSUE TITLE
Integration 

### ISSUE THREAD

**Issue by Taherismail17** [zhuhaow/tun2socks#9]
How to integrate into https://github.com/sheng/ladder-ios

**Comment by zhuhaow** [comment]
This project is not stable, it crashes.

How to do it should be straight forward just as you use any library. If you don’t know what to do with this library, you should learn network first. There is no general rule or anything let you do it in a line of magical code, it depends on what you want to achieve.

**Comment by Taherismail17** [comment]
Hey, I have a Shadowsocks client which will tunnel certain apps and all websites. It doesn’t work with Facebook, Instagram etc. I want to tunnel these websites too similar to Potatso and ShadowRocket. 

Sorry, I’m new to working with VPN Extensions and Shadowsocks

**Comment by zhuhaow** [comment]
There is really no simple way to do it. All the library just help you do some part (usually the most complicated part) of it. But yourself still have to make a lot of decisions. It sounds easy to route traffic from app that does not feel good about just using the system build-in network utility for obviously no good reasons, but there is no general rule and involves really a lot of stuffs (sometimes dirty and hacky) than setting up a proxy server.

Basically, NEKit already provide most things you need, if you wonder how to do it, you can check the code. However, since NEKit depends on tun2socks to processing IP packet data which is not stable, you can't use it to do it yet. But is should provide all the things you need to know if you read everything in NEKit.

**Comment by Taherismail17** [comment]
Thanks for your reply. I will see what I can do. I’m 16 so don’t have too much experience with this yet. 

Thanks
